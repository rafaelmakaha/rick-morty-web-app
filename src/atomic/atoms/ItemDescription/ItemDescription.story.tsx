import React from 'react'
import { Meta, Story } from '@storybook/react'
import ItemDescription, { IItemDescription } from '.'

export default {
  title: 'Atoms/ItemDescription'
} as Meta

const Template: Story<IItemDescription> = (args) => <ItemDescription {...args} />

export const Normal = Template.bind({})

const argTypes = {
    children: {
        control: 'text'
    }
}

const defaultArgs: Omit<IItemDescription> = {}

Normal.args = { ...defaultArgs }
Normal.argTypes = argTypes

