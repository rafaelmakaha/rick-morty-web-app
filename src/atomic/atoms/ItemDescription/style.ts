import styled from 'styled-components'

export const Text = styled.p`
    margin: 4px 0px
`
