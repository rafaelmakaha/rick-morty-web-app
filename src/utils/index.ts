export * from './assignTestId'
export * from './withProviders'
