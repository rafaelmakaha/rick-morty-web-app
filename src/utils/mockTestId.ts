type componentType = 'div' | 'text' | 'input' | 'button' | 'img'

export const assignTestId = (
  componentType: componentType,
  componentId: string
) =>
  process.env.NODE_ENV !== 'production'
    ? {
        'data-testid': `${componentType}_${componentId}`
      }
    : {}

export const mockTestID = (
  componentType: componentType,
  componentId: string
): string => assignTestId(componentType, componentId)['data-testid'] ?? ''
