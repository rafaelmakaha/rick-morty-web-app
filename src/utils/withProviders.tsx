import React from 'react'

import { ThemeProvider } from 'styled-components'

import { theme } from '../tokens'

export const StorybookProviders: React.FC = ({ children }) => (
  <ThemeProvider theme={theme}>{children}</ThemeProvider>
)

export const providerDecorator = (Story: any) => (
  <StorybookProviders>
    <Story />
  </StorybookProviders>
)
