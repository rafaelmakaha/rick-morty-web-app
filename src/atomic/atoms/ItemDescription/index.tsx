import React from "react"
import { Text } from "./style"

export interface IItemDescription {
    testID?: string,
    children: React.ReactNode
}

const ItemDescription: React.FC<IItemDescription> = ({ testID = 'ItemDescription', children}) => (
    <Text>
        {children}
    </Text>
)

export default ItemDescription