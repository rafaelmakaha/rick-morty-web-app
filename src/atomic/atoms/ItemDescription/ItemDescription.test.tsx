import { cleanup, render } from '@testing-library/react'
import React from 'react'
import ItemDescription, { IItemDescription } from '.'
import { mockTestID } from '../../../utils'

describe('Atoms/ItemDescription', () => {
  const props: IItemDescription = {
    testID: 'ItemDescription',
    children: 'SomeChildren'
  }

  const renderComponent = () => render(<ItemDescription {...props} />)

  let component = renderComponent()

  beforeEach(() => {
    jest.clearAllMocks()
    cleanup()
    component = renderComponent()
  })

  const elContainer = mockTestID('div', props.testID!)

  it(`should render component #${elContainer}`, () => {
    const sut = component.findByTestId(el_container)
    expect(sut).toBeTruthy()
  })

  it('should render snapshot', () => {
    expect(component.container.firstChild).toMatchSnapshot()
  })
})
