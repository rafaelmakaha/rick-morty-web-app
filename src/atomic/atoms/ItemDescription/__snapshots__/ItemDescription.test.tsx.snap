// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Atoms/ItemDescription should render snapshot 1`] = `
<p
  class="style__Text-sc-bm4pk8-0 kktcfB"
>
  SomeChildren
</p>
`;
